src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,452 @@
+                    hplDatFile, @"([0-9]+\s+)+# of panels in recursion", $"1   # of panels in recursion", cancellationToken);
+
+            await this.fileSystem.File.ReplaceInFileAsync(
+                    hplDatFile, @"([0-9]+\s+)+NDIVs", $"2   NDIVs", cancellationToken);
+
+            await this.fileSystem.File.ReplaceInFileAsync(
+                    hplDatFile, @"([0-9]+\s+)+# of recursive panel fact.", $"1   # of recursive panel fact.", cancellationToken);
+
+            await this.fileSystem.File.ReplaceInFileAsync(
+                    hplDatFile, @"([0-9]+\s+)+RFACTs", $"2   RFACTs", cancellationToken);
+
+        }
+
+        private async Task DeleteFileAsync(string filePath)
+        {
+            if (this.systemManagement.FileSystem.File.Exists(filePath))
+            {
+                await this.systemManagement.FileSystem.File.DeleteAsync(filePath)
+                    .ConfigureAwait(false);
+            }
+
+        }
+
+        private async Task<string> ExecuteCommandAsync(string pathToExe, string commandLineArguments, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string output = string.Empty;
+
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                this.Logger.LogTraceMessage($"Executing process '{pathToExe}' '{commandLineArguments}' at directory '{workingDirectory}'.");
+                EventContext relatedContext = telemetryContext.Clone()
+                .AddContext("executable", pathToExe)
+                .AddContext("commandArguments", commandLineArguments);
+
+                await this.Logger.LogMessageAsync($"{nameof(HPLinpackExecutor)}.ExecuteWorkload", relatedContext, async () =>
+                {
+                    DateTime start = DateTime.Now;
+
+                    using (IProcessProxy process = this.systemManagement.ProcessManager.CreateProcess(pathToExe, commandLineArguments, workingDirectory))
+                    {
+                        SystemManagement.CleanupTasks.Add(() => process.SafeKill());
+
+                        await process.StartAndWaitAsync(cancellationToken).ConfigureAwait(false);
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            this.Logger.LogProcessDetails<HPLinpackExecutor>(process, telemetryContext);
+                            process.ThrowIfErrored<WorkloadException>(ProcessProxy.DefaultSuccessCodes, errorReason: ErrorReason.WorkloadFailed);
+                        }
+
+                        output = process.StandardOutput.ToString();
+
+                    }
+                }).ConfigureAwait(false);
+            }
+
+            return output;
+        }
+
+        private async Task<string> ExecuteElevatedCommandAsync(string pathToExe, string commandLineArguments, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken)
```

Comment:
Just pass in an optional boolean to the ExecuteCommandAsync method above. No need for 2 methods.

private async Task<string> ExecuteCommandAsync(string pathToExe, string commandLineArguments, string workingDirectory, EventContext telemetryContext, CancellationToken cancellationToken, bool runElevated = false)